wgsl/index.bs

```diff
@@ -4557,11 +4555,271 @@ TODO: *Stub*: Expression evaluation
 
 ## Uniformity TODO ## {#uniformity}
 
-### Uniform control flow TODO ### {#uniform-control-flow}
+### Terminology and concepts TODO ### {#uniformity-concepts}
+
+The following definitions are merely informative, trying to give an intuition for what the analysis in the next subsection is computing.
+The analysis is what actually defines these concepts, and when a program is valid or break the uniformity rules.
+
+Control-flow is said to be uniform at a point in the program if all invocations execute in lockstep at that program point.
```

Comment:
For purposes of utility, I would define uniform control flow as being control flow that depends on values certain to be the same over the defined scope of uniformity. The programmer doesn't care if they are executed in lockstep or not. It's not strictly relevant to the restrictions this introduces. Rather than having to define how hardware works, let's define how you can write a program that trips it up.